table/tables/tables.go

```diff
@@ -406,11 +389,26 @@ func (t *Table) rebuildIndices(rm kv.RetrieverMutator, h int64, touched map[int]
 }
 
 // AddRecord implements table.Table AddRecord interface.
-func (t *Table) AddRecord(ctx context.Context, r []interface{}, h int64) (recordID int64, err error) {
-	// Already have recordID
-	if h != 0 {
-		recordID = int64(h)
-	} else {
+func (t *Table) AddRecord(ctx context.Context, r []interface{}) (recordID int64, err error) {
+	var hasRecordID bool
+	for _, col := range t.Cols() {
+		if mysql.HasPriKeyFlag(col.Flag) && t.meta.PKIsHandle {
+			// The value must have been filled with int64 or uint64 value, or it will not pass null check.
+			switch x := r[col.Offset].(type) {
+			case int64:
+				recordID = x
+			case uint64:
+				recordID = int64(x)
+			case int:
```

Comment:
The value has been cast to the column type, so it is int64 or uint64, but I should use `types.ToInt64` for simplicity, just need to optimize it a little bit.